pkg/crypto/crypto_test.go

```diff
@@ -126,6 +128,49 @@ func TestEncryptDecrypt(t *testing.T) {
 	})
 }
 
+func TestUpgradePathEncryptDecrypt(t *testing.T) {
```

Comment:
Shouldn't this be `TestDecryptFromJSONString()`? The upgrade path is just one of the test cases.

Reply:
If that's the case why don't you add this as a separate test case to the existing method?

Reply:
So you are adding an encrypt test method and taking the output of that and testing DecryptFromJSONString(). IOW, you are really not testing any `encrypt` functionality. So `TestUpgradePathDecrypt()` might be more appropriate. However if we continue to add more unit test cases for `DecryptFromJSONString()`, they all should be collapsed into a `TestDecryptFromJSONString()` method.

Reply:
Yes. TestUpgradePathDecrypt more appropriate one. 
agreed to the second comment. will change the test function name back to original .